client/src/fluid/FluidAutocomplete.ml

```diff
@@ -329,16 +329,57 @@ let secretToACItem (s : SecretTypes.t) : fluidAutocompleteItem =
   FACVariable (s.secretName, Some asDval)
 
 
+let lookupIsInQuery (tl : toplevel) ti functions =
+  let queryFnNames =
+    functions
+    |> List.map ~f:(fun fn ->
+           match fn with
+           | FACFunction fn ->
+               if fn.fnIsQuery then Some fn.fnName else None
+           | _ ->
+               None)
+    |> List.filter ~f:Option.is_some
+    |> List.map ~f:(fun x -> match x with Some fnName -> fnName | _ -> "")
+  in
+  let isQueryFn name = queryFnNames |> List.any ~f:(fun q -> q = name) in
```

Comment:
This works, it's probably fine, but I don't like it!

The AST's EFnCall has the function name, but doesn't have the actual fn record (I don't think?). So we have to look up the fn record to check the fnIsQuery field. I did this by filtering down to the query functions first, then checking each ancestor function name against that list. So at least we're not searching the entire functions list for each ancestor name. But, we're still filtering down all the functions to query functions every time we generate the autocomplete list. I didn't figure out a place to do that just once.

Reply:
OK, I think you're right here. Let's go back to the old way.

I plan to add the fn to the EFnCall at some point, but the old way is certainly simpler until that happens.

Reply:
reverted!